pkg/git/cache.go

```diff
@@ -0,0 +1,149 @@
+package git
+
+import (
+	"log"
+	"os"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+)
+
+// set to override cache dir
+var CacheDir = os.Getenv("HOME") + "/.anyvendor/git"
+
+// set to override progress logging
+var ProgressOut = os.Stdout
+
+// GitVendorCache maintains a local cache of vendored git repos
+type GitVendorCache struct {
+	Dir string
+}
+
+func DefaultCache() *GitVendorCache {
+	return &GitVendorCache{Dir: CacheDir}
+}
+
+func (c *GitVendorCache) Init() error {
+	return os.MkdirAll(c.Dir, 0777)
+}
+
+func (c *GitVendorCache) EnsureCheckedOut(
+	url, sha, tag, authUser, authToken string,
+) error {
+	repoDir, _ := c.GetRepoDir(url)
+	repoExists, err := fileExists(repoDir)
+	if err != nil {
+		return err
+	}
+	var authMethod transport.AuthMethod
+	if authToken != "" {
+		authMethod = &http.BasicAuth{Username: authUser, Password: authToken}
```

Comment:
How would this work with a GITHUB_TOKEN in CI?

Reply:
```go
AuthUser:  os.Getenv("GITHUB_USER"),  
AuthToken:  os.Getenv("GITHUB_TOKEN"),
```

Reply:
`GITHUB_TOKEN` is all you need